HashTagTokenizer.java

```diff
@@ -29,12 +36,20 @@ public static void breakHashTag(String hashtag, String[] dictionary) {
         if (hashtag.isEmpty()) {
             return;
         }
- 
+		hashtag = hashtag.toLowerCase();
         int N = hashtag.length();
+		int endIndex = 1;
```

Comment:
Why do you need `endIndex`? use `i` they both always holding same value within your loop. 
```suggestion
```